src/nomad_analysis/jupyter/schema.py

```diff
@@ -410,178 +338,216 @@ def write_predefined_cells(
+            '    background-position: left bottom, right top;\n',
+            '    background-repeat: no-repeat,  no-repeat;\n',
+            '    background-size: auto 60px, auto 160px;\n',
+            '    border-radius: 5px;\n',
+            '    box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0,0,0,.12);">\n',  # noqa: E501
+            '\n',
+            '<h1 style="\n',
+            '    color: #2a4cdf;\n',
+            '    font-style: normal;\n',
+            '    font-size: 2.25rem;\n',
+            '    line-height: 1.4em;\n',
+            '    font-weight: 600;\n',
+            '    padding: 30px 200px 0px 30px;"\n',
+            f'>{self.name}</h1>\n',
+            '<p style="font-size: 1.25em; font-style: italic; padding: 5px 200px 30px 30px;"\n',  # noqa: E501
+            f'>{user}</p>\n',
+            '</div>\n',
+            '\n',
+            'This notebook has been generated by a NOMAD Analysis entry with the\n',
+            'following definition path:\n',
+            f'`{self.m_to_dict()["m_def"]}`',
```

Comment:
I get a key error when running this in the Python REPL:
```
>>> test = JupyterAnalysis()
>>> test.m_def
nomad_analysis.jupyter.schema.JupyterAnalysis:Section
>>> test.m_to_dict()["m_def"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'm_def'
```

Reply:
I think this might be a better solution:

```suggestion
            f'`{self.m_def.qualified_name()}`',
```

```
>>> test.m_def.qualified_name()
'nomad_analysis.jupyter.schema.JupyterAnalysis'
```

Reply:
I think `self` points to the EntryArchive which should have `m_def`

Reply:
I'm pretty sure `self` should point to the section. Otherwise there would be no need to supply archive argument to the normalize method.

Reply:
Hmm, I am pretty sure this was working when I tested before. Good catch, and your suggestion above works as well, better than deserializing anyways

Reply:
I just checked, `self.m_to_dict()["m_def"]` works with the `ServerContext`. Nevertheless I like your way better.

Reply:
Ah, okay. That explains it. But yes, this should be a bit more efficient than serializing the whole section.